[docs-infra] Remove explicit `mui-x` dependency resolving

LukasTy opened the pull request:
- Remove explicit `mui-x` relevant packages resolving in monorepo relevant code
- introduce the optional `postProcessImport` "hook", with which mui-x can post-process the import on it's end
- Remove `mui-x` related tests in favour of https://github.com/mui/mui-x/pull/11463/commits/f67a686f2fb486c819158d806d4e2adc47d2d9d0

Needed to resolve time-sensitive issue due to breaking `date-fns` v3 release: https://github.com/mui/mui-x/pull/11463

flaviendelangle reviewed (approved):
Looks good to me, this logic will be a lot easier to maintain on our repo

Just, are we sure that we don't have any demo on the core doc that depends on it?

LukasTy commented:
> Just, are we sure that we don't have any demo on the core doc that depends on it?

Everything works as expected unless I couldn't find some demo related to Data Grid or Pickers. 🙈 
There are basically no demos in docs having X dependencies and an option to edit it in codesandbox, hence, nothing seems impacted.
I've [reverted one change](https://github.com/mui/material-ui/pull/40257/commits/bed98a3b2123b992bd0541f1bcb0133741418b95) that could have had the slightest impact, because there is a demo using Data Grid that is editable. 👍

oliviertassinari commented:
I changed the labels to be docs-infra. This looks 100% about having the proper abstraction level, it was a leaky abstraction before.

LukasTy commented:
> I changed the label to be `docs-infra`. This looks about having the proper abstraction level, it was a leaky abstraction before.

Thanks, in that case, I've asked for a review from `docs-infra`. 👍 

> What about this demo though https://mui.com/x/react-date-pickers/getting-started/#render-your-first-component? I guess it's not needed because the dependencies are hardcoded.

It's the same as for all the other demos. [We moved](https://github.com/mui/mui-x/pull/11463/files#diff-ac973c91d8ec203f9e89a58407e831f492da764c2f675587630bd52436286b97) the relevant related dependency resolving into `mui-x` to have better control over it and avoid fighting the logic existing in "monorepo".

LukasTy commented:
> [!WARNING]
> This branch should not be deleted, because it is used for [v6 (master)](https://github.com/mui/mui-x/blob/master/package.json#L88) and will remain being used for `docs-v6`, even after `v7` becomes stable.
